apollo-ios-codegen/Sources/ApolloCodegenLib/Frontend/JavaScript/src/validationRules.ts

```diff
@@ -89,39 +88,23 @@ export function DeferredInlineFragmentNoTypeCondition(context: ValidationContext
   };
 }
 
-export function DeferDirectiveMissingLabelArgument(context: ValidationContext) {
-  function ValidateDeferDirectiveLabelArgument(node: InlineFragmentNode | FragmentSpreadNode, error: GraphQLError) {
-    if (node.directives) {
-      for (const directive of node.directives) {
-        if (directive.name.value == GraphQLDeferDirective.name && !(directive.arguments?.find(
-          function(element) {
-            return element.name.value == 'label'
-          }) ? true: false)
-        ) {
-          context.reportError(error)
-        }
-      }
-    }
-  }
-
+export function DeferredInlineFragmentMissingLabelArgument(context: ValidationContext) {
   return {
     InlineFragment(node: InlineFragmentNode) {
-      ValidateDeferDirectiveLabelArgument(
-        node, 
-        new GraphQLError(
-          "Apollo requires all @defer directives to use the 'label' argument. Please add a 'label' argument to the @defer directive on this inline fragment.",
-          { nodes: node }
-        )
-      )
-    },
-    FragmentSpread(node: FragmentSpreadNode) {
-      ValidateDeferDirectiveLabelArgument(
-        node,
-        new GraphQLError(
-          "Apollo requires all @defer directives to use the 'label' argument. Please add a 'label' argument to the @defer directive on this fragment spread.",
-          { nodes: node }
-        )
-      )
+      if (node.directives) {
+        for (const directive of node.directives) {
+          if (directive.name.value == GraphQLDeferDirective.name && !(directive.arguments?.find((element) =>
+            element.name.value == 'label'
+          ))) {
+            context.reportError(
+              new GraphQLError(
+                "Apollo requires all @defer directives to use the 'label' argument. Please add a 'label' argument to the @defer directive on this inline fragment.",
```

Comment:
Should we also change the error message to reflect this new logic?

Reply:
I've updated it in https://github.com/apollographql/apollo-ios-dev/pull/33/commits/9f0dddd12a84c5033892a64a03bc62bf6618dce9.